lib/OpenQA/Schema/ResultSet/JobSettings.pm

```diff
@@ -28,11 +29,28 @@ sub jobs_for_setting ($self, $options) {
     my $max_id = defined $result ? $result->[0] : 0;
     my $min_id = $max_id > $limit ? $max_id - $limit : 0;
 
+    my @where_queries = ('js.key LIKE ?', 'js.value LIKE ?', 'js.job_id > ?');
+    my @where_values = ($key_like, $list_value_like, $min_id);
+    my $join_part = '';
+    if (defined $list_result) {
+        my $list_result_like = "%${list_result}%";
+        $join_part = 'JOIN jobs j ON js.job_id = j.id';
+        push @where_queries, 'j.result LIKE ?';
+        push @where_values, $list_result_like;
+    }
     # Instead of limiting the number of jobs, this query could also use a trigram gin index to achieve even better
     # performance (at the cost of disk space and setup complexity)
-    $sth = $dbh->prepare(
-        'SELECT job_id, value FROM job_settings WHERE key LIKE ? AND value LIKE ? AND job_id > ? ORDER BY job_id DESC');
-    $sth->execute($key_like, $list_value_like, $min_id);
+    my $where_queries = 'WHERE ' . join(' AND ', @where_queries);
+    my $sql = "SELECT js.job_id, js.value
+               FROM job_settings js
+               $join_part
+               $where_queries
+               ORDER BY js.job_id DESC";
+    $sth = $dbh->prepare($sql);
+    $sth->execute(@where_values);
```

Comment:
It looks like we only have two cases here (`$list_result` is defined or not). So maybe it would be clearer to just spell out the entire SQL query in one piece for each case (despite the repetition). It would be easier to read and less likely contain any chances for SQL injections.

Reply:
But I'm wondering why this is using raw SQL instead of DBIx::Class features? Is there anything in the query not supported by DBIC?

Reply:
When trying optimizations to search in a comma-separated list (the real optimization would of course be to use a proper database schema) I noticed that DBIx gets more in the way than it helps. So maybe this question is not leading in a useful direction.

But to answer the question: I would say DBIx is also not making things really better here, too. Otherwise the question probably goes to @kraih who wrote this query.

Reply:
Please note that the common abbreviation for DBIx::Class is "DBIC". "DBIx" is a namespace that contains many other DBI related modules that have nothing to do with ORMs.

Reply:
Afraid i don't remember anymore why this is a raw query. But there probably was a reason. Maybe DBIx::Class generated a less optimal query? I do remember performance being a big concern here.